cdi/src/main/java/org/hibernate/validator/cdi/internal/ValidationExtension.java

```diff
@@ -6,416 +6,9 @@
-
-		if ( executableTypes.contains( ExecutableType.IMPLICIT ) ) {
-			if ( isGetter ) {
-				executableTypes.add( ExecutableType.GETTER_METHODS );
-			}
-			else {
-				executableTypes.add( ExecutableType.NON_GETTER_METHODS );
-			}
-		}
-
-		return executableTypes;
-	}
-
-	private EnumSet<ExecutableType> executableTypesDefinedOnConstructor(Constructor<?> constructor) {
-		ValidateOnExecution validateOnExecutionAnnotation = constructor.getAnnotation(
-				ValidateOnExecution.class
-		);
-		EnumSet<ExecutableType> executableTypes = commonExecutableTypeChecks( validateOnExecutionAnnotation );
-
-		if ( executableTypes.contains( ExecutableType.IMPLICIT ) ) {
-			executableTypes.add( ExecutableType.CONSTRUCTORS );
-		}
-
-		return executableTypes;
-	}
-
-	private EnumSet<ExecutableType> commonExecutableTypeChecks(ValidateOnExecution validateOnExecutionAnnotation) {
-		if ( validateOnExecutionAnnotation == null ) {
-			return EnumSet.noneOf( ExecutableType.class );
-		}
-
-		EnumSet<ExecutableType> executableTypes = EnumSet.noneOf( ExecutableType.class );
-		if ( validateOnExecutionAnnotation.type().length == 0 ) {  // HV-757
-			executableTypes.add( ExecutableType.NONE );
-		}
-		else {
-			Collections.addAll( executableTypes, validateOnExecutionAnnotation.type() );
-		}
-
-		// IMPLICIT cannot be mixed 10.1.2 of spec - Mixing IMPLICIT and other executable types is illegal
-		if ( executableTypes.contains( ExecutableType.IMPLICIT ) && executableTypes.size() > 1 ) {
-			throw log.getMixingImplicitWithOtherExecutableTypesException();
-		}
-
-		// NONE can be removed 10.1.2 of spec - A list containing NONE and other types of executables is equivalent to a
-		// list containing the types of executables without NONE.
-		if ( executableTypes.contains( ExecutableType.NONE ) && executableTypes.size() > 1 ) {
-			executableTypes.remove( ExecutableType.NONE );
-		}
-
-		// 10.1.2 of spec - A list containing ALL and other types of executables is equivalent to a list containing only ALL
-		if ( executableTypes.contains( ExecutableType.ALL ) ) {
-			executableTypes = ALL_EXECUTABLE_TYPES;
-		}
-
-		return executableTypes;
-	}
-
-	public Method replaceWithOverriddenOrInterfaceMethod(Method method, List<Method> allMethodsOfType) {
-		LinkedList<Method> list = new LinkedList<>( allMethodsOfType );
-		Iterator<Method> iterator = list.descendingIterator();
-		while ( iterator.hasNext() ) {
-			Method overriddenOrInterfaceMethod = iterator.next();
-			if ( executableHelper.overrides( method, overriddenOrInterfaceMethod ) ) {
-				if ( method.getAnnotation( ValidateOnExecution.class ) != null ) {
-					throw log.getValidateOnExecutionOnOverriddenOrInterfaceMethodException( method );
-				}
-				return overriddenOrInterfaceMethod;
-			}
-		}
-
-		return method;
-	}
+@Deprecated
+public class ValidationExtension extends org.hibernate.validator.cdi.ValidationExtension {
```

Comment:
I'm wondering why this needs exporting at all? I've asked Arjan on the issue, let's see what he says.